### ISSUE TITLE
Spring boot application deployment in Liberty throwing Class cast exception

### ISSUE THREAD

**Issue by ghost** [OpenLiberty/open-liberty#4310]
Please find the link to my issue in Stack overflow.

https://stackoverflow.com/questions/51383728/spring-boot-application-deploymnet-in-liberty-throwing-class-cast-exception?noredirect=1#comment89779158_51383728

Please find the attached source code.

[smartdemo.zip](https://github.com/OpenLiberty/open-liberty/files/2206188/smartdemo.zip)

**Comment by tjwatson** [comment]
The smartdemo pom.xml has many references to the class com.demo.smartdemo.RxSmartDemoLauncher which does not exist.  If I change them all to com.demo.smartdemo.SmartDemoLauncher (by removing `Rx`) then the application starts fine for me on open-liberty 18.0.0.1.  Can you also post the complete server.xml file you are using.  Also I assume you are using open-liberty 18.0.0.1 as downloaded from https://public.dhe.ibm.com/ibmdl/export/pub/software/openliberty/runtime/release/2018-06-19_0502/openliberty-18.0.0.2.zip

**Comment by abrahamjins** [comment]
Thanks Tom for the feedback.
We are using the Liberty Core 18.0.0.1. Can you confirm below points?
1. does Liberty Core 18.0.0.1 support Spring boot deployment(war)? If not is there any workaround to achieve this?
2.We have tried to deploy spring boot war as a web application. It was failing and we are able to fix it by removing application.xml and move corresponding configuration to server.xml file.It was a mandate by the client that giving application specific configuration only in application.xml.
The issue is that spring is actually taking this liberty application.xml for initilizing the spring beans instead of the configuration specified in the spring  boot. How can we tell Spring to not looking at liberty application.xml?

Attaching th eserver.xml below.
<server description="Anthem">
    <featureManager>
        <feature>jsp-2.2</feature>
        <feature>localConnector-1.0</feature>
        <feature>servlet-3.1</feature>
    </featureManager>
	
	
	
	<variable name="RXSMART" value="server1"/>
	<variable name="HOST" value="localhost"/>
        <variable name="wmqJmsClient.rar.location" value="${shared.resource.dir}/mq/wmq.jmsra.rar"/>
	
	 
	<featureManager>
		<feature>adminCenter-1.0</feature>
	</featureManager>
	<quickStartSecurity userName="admin" userPassword="passw0rd"/>

	  
	<httpEndpoint host="*" httpPort="9090" httpsPort="9443" id="defaultHttpEndpoint">
		<sslOptions sslRef="anthemSSLSettings"/>
	</httpEndpoint>

   
	
	<logging traceSpecification="myLogger=FINEST"/>

    <applicationManager autoExpand="true"/>
	<applicationMonitor dropins="dropins" dropinsEnabled="true" updateTrigger="mbean"/>

    <webApplication id="rxsmart" contextRoot ="rxsmart" location="provider.war" name="provider"/>
</server>

**Comment by tjwatson** [comment]
Thanks @scottkurz I have been able to reproduce the issue.  It is happening because of the following FFDC:

```
java.lang.ClassCastException: com.ibm.ws.app.manager.springboot.internal.SpringBootModuleInfo cannot be cast to com.ibm.ws.container.service.app.deploy.WebModuleInfo
        at com.ibm.ws.ejbcontainer.osgi.internal.ModuleInitDataAdapter.adapt(ModuleInitDataAdapter.java:100)
        at com.ibm.ws.ejbcontainer.osgi.internal.ModuleInitDataAdapter.adapt(ModuleInitDataAdapter.java:40)
        at com.ibm.ws.adaptable.module.internal.AdapterFactoryServiceImpl.adapt(AdapterFactoryServiceImpl.java:196)
        at com.ibm.ws.adaptable.module.internal.AdaptableContainerImpl.adapt(AdaptableContainerImpl.java:170)
        at com.ibm.ws.ejbcontainer.osgi.internal.EJBContainerImpl.createEJBInWARModuleMetaData(EJBContainerImpl.java:143)
        at com.ibm.ws.ejbcontainer.war.internal.EJBWARMetaDataRuntime.createdNestedModuleMetaData(EJBWARMetaDataRuntime.java:63)
        at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase$ModuleContainerInfoBase.createModuleMetaData(SimpleDeployedAppInfoBase.java:173)
        at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.preDeployApp(DeployedAppInfoBase.java:368)
        at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.deployApp(DeployedAppInfoBase.java:412)
        at com.ibm.ws.app.manager.springboot.internal.SpringBootHandler.install(SpringBootHandler.java:135)
        at com.ibm.ws.app.manager.internal.statemachine.StartAction.execute(StartAction.java:140)
        at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.enterState(ApplicationStateMachineImpl.java:1258)
        at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.run(ApplicationStateMachineImpl.java:873)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

It can be worked around by excluding the `webProfile-8.0` feature or anything that pulls in the EJB support.

@abrahamjins when this fails for you can you look in the <server config>/logs/ffdc folder for any error logs and please attach them here.  I have yet to reproduce this with the feature set you have indicated is enabled.

**Comment by schriste** [comment]
I ran into this same issue locally, following your advice I fixed it by commenting out the javaee-7.0 feature
    <featureManager>
         
        <feature>concurrent-1.0</feature>
        <feature>localConnector-1.0</feature>
        <feature>servlet-3.1</feature>
        <feature>springBoot-2.0</feature>
    </featureManager>

This is a showstopper for me, as I need to deploy both Spring Boot and EJB apps.  Is there no option to do this currently?

**Comment by abrahamjins** [comment]
@tjwatson,

I was able to deploy it with all configuration in server.xml / rename application.xml to some other name.
I have deployed my app as springBootApplication and all looks good. 

Now my task is to adhere to the configuration suggested by client, that use application.xml for app specific configuration. when the xml is named as application.xml, spring considers it for spring component initialization.That should not happen. Is there any way to restrict this?
